kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/engine/DefaultCommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.engine;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import io.delta.storage.LogStore;
+import io.delta.storage.commit.CommitCoordinatorClient;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.CommitCoordinatorClientHandler;
+import io.delta.kernel.engine.coordinatedcommits.CommitResponse;
+import io.delta.kernel.engine.coordinatedcommits.GetCommitsResponse;
+import io.delta.kernel.engine.coordinatedcommits.UpdatedActions;
+import io.delta.kernel.engine.coordinatedcommits.actions.AbstractMetadata;
+import io.delta.kernel.engine.coordinatedcommits.actions.AbstractProtocol;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.defaults.internal.coordinatedcommits.CommitCoordinatorProvider;
+import io.delta.kernel.defaults.internal.coordinatedcommits.StorageKernelAPIAdapter;
+import io.delta.kernel.defaults.internal.json.JsonUtils;
+import io.delta.kernel.defaults.internal.logstore.LogStoreProvider;
+
+/**
+ * Default implementation of {@link CommitCoordinatorClientHandler} based on Hadoop APIs which uses
+ * commit coordinator defined in delta-storage modules.
+ * It takes a Hadoop {@link Configuration} object to interact with the commit coordinator client.
+ * The following optional configurations can be set to customize the behavior of the client:
+ * <ul>
```

Comment:
do we want to document the details about configuring the commit coordinator provider? `io.delta.kernel.commitCoordinatorBuilder.*"`?